### DISCUSSION TITLE
Stream Queue Size Increases Rapidly

### DISCUSSION THREAD

**Discussion by ahaverty** [fuatakgun/eufy_security#575] [Q&A]
My "stream queue size" increases quite fast.
I had a look around but couldn't find any info on what this attribute means.
Is that normal, or a sign of issues with my configuration?
<img width="348" alt="image" src="https://user-images.githubusercontent.com/2920947/207168987-26522758-0b5d-4379-8ff6-2d58a3fed406.png">

I'm on a wired 2k doorbell, with P2P and webrtc

**Comment by fuatakgun** [comment]
After each start of p2p stream, this should go to 0.

Incoming data bytes are forwarded to ffmpeg process to generate video on the fly.

If the number goes high, either ffmpeg process is broken (needs to be recovered) or hardware is not able to cope with conversion.

1. Is streaming working?
2. Did you set all quality stuff to low?

**Reply by ahaverty** [reply]
@fuatakgun Thanks for responding 👏 

> 1. Is streaming working?

Less than 50% of the time, usually needs eufy add-on/integration reboots

> 2. Did you set all quality stuff to low?

Yes, but all I can find is `Video Settings > Video Quality > Streaming Quality > Low` (I can't see any codec settings for my 2k wired doorbell.)

> ffmpeg process is broken (needs to be recovered)

Would you have any tips on how to do this?

> hardware is not able to cope with conversion

My HA Blue CPU is <10% while the doorbell is/isn't streaming, and memory <50%. Would I be right in ruling hardware performance out?

**Context that might be useful:**
Streaming works sometimes (less than 50% of the time).
Streaming via WebRTC works percectly everytime I go to fix/debug the stream, by reloading add-ons, and triggering services.
But hours or a day later, I discover it's erroring when I actually need it, usually something like this:
<img width="414" alt="image" src="https://user-images.githubusercontent.com/2920947/207672005-7e6f2d17-a533-4d0c-8862-a35e3824cc4e.png">

I don't have a full list of different configurations I've tried, but I have tried following the Readme exactly, multiple times + troubleshooting guide, and I'm starting to think it's not as applicable to the 2K wired doorbell(?).

You're a saint for creating/maintaining these repo's, I don't want to be bothering you with a load of issues, but if you know of anyone with a 2k wired doorbell, working with this repo, perhaps they could share their config? _I'd asked awhile ago, but didn't have much luck_: https://github.com/fuatakgun/eufy_security/discussions/501

Cheers!

**Comment by fuatakgun** [comment]
I am working on revamping the architecture while waiting new version of add-on to be released. Let's agree that, maybe this weekend or next weekend, you can share your camera and debug the issue. I am sure there are many out there having same similar issues.

**Comment by ahaverty** [comment]
Thanks @fuatakgun , that should suit 👌 

I've glossed over the Coded instructions as I couldn't find them in my settings, but I see here @olivierdj and @aaronganek were able to 
set their codec in the eufy app https://github.com/fuatakgun/eufy_security/issues/126#issuecomment-1010258493
<img src="https://user-images.githubusercontent.com/2920947/207686599-c49d1749-0ce1-469d-8335-a0fb574b9254.jpeg" width="220"><img src="https://user-images.githubusercontent.com/2920947/207686594-6325e453-86d1-4478-80a2-d4fc91e5c485.jpeg" width="220">

That's all the options I have. I wonder have Eufy changed access, or maybe it's a difference of regions (I'm in Ireland).
Maybe if they see this, they could help me understand how they changed codec settings.

**Comment by ahaverty** [comment]
Just for reference, to show that hardware seems very acceptable even while the queue size is shooting up:

https://user-images.githubusercontent.com/2920947/207918430-5080346d-d592-4db2-acca-616ba786ee12.mov

**Comment by fuatakgun** [comment]
@ahaverty  please give it a try with Alpha version

**Comment by ahaverty** [comment]
Thanks @fuatakgun, I've got the latest beta setup now.
I see the queue size is rapidly incrementing from 0 to about a max of 8, before resetting back to 0.
Other than a short white spinner every few seconds, I don't notice any issues with the actual stream, while viewing it through webrtc card.
Does that sound correct to you?
<img width="355" alt="image" src="https://user-images.githubusercontent.com/2920947/210285822-bf0201ff-6d59-48c7-9214-fe94ab74728b.png">

**Reply by fuatakgun** [reply]
Yes all good

**Comment by ahaverty** [comment]
Just to give a sense of how quickly it's swapping the queue size, here's the history for 5 minutes:
<img width="2741" alt="image" src="https://user-images.githubusercontent.com/2920947/210286761-3ddda033-212d-4030-8555-141575e6192d.png">

I wonder could that make a mess of my database, all those writes? 🤔 

My HA Blue system seems happy enough, while it's streaming, for what its worth:
<img width="669" alt="image" src="https://user-images.githubusercontent.com/2920947/210286837-e1fcd52a-8ac8-4fc7-bff9-804b3fecc213.png">

**Reply by fuatakgun** [reply]
Feel free to disable it

**Reply by ejpenney** [reply]
Just chiming in here that disabling this particular debug sensor has made HUGE improvements to the performance of my 2k Dual Doorbell streams.  Streams come up faster (~5 seconds vs the previous 30-40 seconds), Also, generally Home Assistant doesn't get bogged down: I was noticing when a stream event was occurring ZigBee devices took 20-30 seconds to respond.  I guess it was all the DB read/writes for this sensor.  Perhaps it shouldn't be enabled by default?

**Reply by fuatakgun** [reply]
I am open to suggestions to keep this disabled by default or hide from the recorder, but i haven't found a solution yet.

**Reply by ejpenney** [reply]
At the risk of displaying how completely ignorant I am to how this particular integration works... Couldn't we just set _attr_entity_registry_enabled_default=False for entities that look like this one?

**Reply by fuatakgun** [reply]
Thanks, i didn't know that, please go ahead and raise a pr if you have time, otherwise, i can do it later on

**Reply by ejpenney** [reply]
Sure, simplest just to do it for all diagnostic entities, like in [this PR](https://github.com/fuatakgun/eufy_security/pull/888).

Change wouldn't affect existing installations, safest since we don't know if they're using these entities or not...
